### ISSUE TITLE
AETTA: Label-Free Accuracy Estimation for Test-Time Adaptation

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#63]
<img width="1422" alt="image" src="https://github.com/user-attachments/assets/682a0fd7-fdc2-423f-b6e1-52a6ae28d6e2">

### Paper Link

https://openaccess.thecvf.com/content/CVPR2024/html/Lee_AETTA_Label-Free_Accuracy_Estimation_for_Test-Time_Adaptation_CVPR_2024_paper.html

### Code Link

https://github.com/taeckyung/AETTA

## Motivation

Although recent TTA studies deal with dynamic test streams in TTA, the inherent risk of TTA blind adaptation to unseen test samples without groundtruth labels–remains a critical vulnerability. Notably, the **absence of ground-truth labels** makes it difficult to monitor the correctness of the adaptation.

<img width="500" alt="image" src="https://github.com/user-attachments/assets/c8693de7-fb40-40b3-82bd-c69339ac84c4">

### PDD

To estimate the accuracy of the model, we propose **prediction disagreement with dropout inferences (PDD)** that calculates a disagreement between the adapted model $h(\cdot; \Theta)$ and the dropout inferences $h(\cdot; \Theta^{dropout})$ with respect to test samples as:

$$
PDD_{D\tau}(h) \triangleq \mathbb{E}_{D\tau} \left[ \frac{1}{N} \sum_{i=1}^{N} 1[h(X; \Theta) \neq h(X; \Theta^{dropout}_i)] \right],
$$

where $N$ is the number of dropout inferences.

We approximate a single model test error as:

$$
Err_{D\tau}(h) \approx b \, PDD_{D\tau}(h)
$$

A desirable $b$ should dynamically suppress the **over-confident** expectation function depending on the context so that the confidence-prediction calibration assumption holds. To this end, we use the **skewness of the predicted outputs** as an indicator of model over-confidence. Our intuition is based on the observation that the predicted class distribution is **highly skewed** when the adaptation **fails** (Figure 2b), which aligns with the findings from prior studies [19, 24].

<img width="500" alt="image" src="https://github.com/user-attachments/assets/8b79b28b-fa9d-42c7-980e-4be04a9da719">

> <img width="312" alt="image" src="https://github.com/user-attachments/assets/f0c2d50e-ffc8-4687-8fb5-b553a4cc204f">

> SAR [ICLR23]: as the adaptation goes by, the model tends to predict all input samples to the same class, even though these samples have different ground-truth classes, called model collapse.

Specifically, we estimate the skewness of predictions by calculating the **entropy** (Ent) of the batch-aggregated softmax values from the dropout inferences over a test batch $X_t$:

$$
E^{avg} = Ent\left( \frac{1}{N} \sum_{i=1}^{N} \frac{1}{|X_t|} \sum_{x \in X_t} f(x; \Theta^{dropout}_i) \right)
$$

where $E^{avg}$ would maximize as $E^{max} = Ent(\bar{1}_K / K)$ with uniform predictions among the batch (e.g., no failures); while the minimum value would be 0 when the entire batch predicts a single class (e.g., adaptation failures).

We then model $b$ with $E^{avg}$ as:

$$
b = \left( \frac{E^{avg}}{E^{max}} \right)^{-\alpha}, \tag{12}
$$

where $\alpha \in [0, \infty)$ is a hyperparameter. If the adaptation does not fail, predictions are uniformly distributed as $E^{avg} = E^{max}$ and $b = 1$. We found that modeling $b$ with the average batch-wise entropy effectively corrects the correlation between confidence and prediction probability, as illustrated in Figure 3 (blue dots).

<img width="1536" alt="image" src="https://github.com/user-attachments/assets/10ded1bc-9957-47d5-88ee-24c770464748">

Finally, we propose **Accuracy Estimation for TTA (AETTA)**:

$$
Err_{D\tau}(h) \approx \left( \frac{E^{avg}}{E^{max}} \right)^{-\alpha} PDD_{D\tau}(h). \tag{13}
$$

<img width="500" alt="image" src="https://github.com/user-attachments/assets/2961e742-73ff-4a32-ad79-2bac66f319d1">

<img width="500" alt="image" src="https://github.com/user-attachments/assets/18a86614-03ce-4f44-9117-5b47b4248889">

## Experiments 

<img width="1094" alt="image" src="https://github.com/user-attachments/assets/19a3b83d-32a9-40f9-9e3a-e8832445a1b5">

<img width="1178" alt="image" src="https://github.com/user-attachments/assets/502efdf8-ee1b-4447-905a-7be653b42295">
